### ISSUE TITLE
How to use StridedMemrefView, or is there an alternative interface？

### ISSUE THREAD

**Issue by mars1248** [openxla/xla#14194]
Last year I followed the implementation logic in this file, https://github.com/openxla/xla/blob/7954169ccfb6290d94af3ea3634229b097682ba8/xla/service/gpu/runtime/gemm.cc
 and the input parameters were defined as StridedMemrefView, but I see that the master code has almost no use for this interface, changed to BufferAllocation::Slice, but my logic needs a raw view, To get the shape of each dimension, what interface should I use?
and  I found use thunk impl custom call, why is this？

**Comment by cheshire** [comment]
Thanks for your report, perhaps it's better to start by describing the problem you'd like to solve?

**Comment by mars1248** [comment]
> Thanks for your report, perhaps it's better to start by describing the problem you'd like to solve?

Thank you for your answer. Last year, I implemented resize bicubic, a custom call. To be used by torch xla's upsampler_bicubic. However, the previous implementation of custom call always passed StridedMemrefView as a parameter, but now it seems to be BufferAllocation::Slice, so we can't get the Strided information of the tensor directly

**Comment by cheshire** [comment]
Ah interesting. I don't think XLA ever supported StridedMemRef (or maybe it pretended to support, but actually ignored the stride). We've just discussed adding support to strided views as a possible feature. CC @ezhulenev @bchetioui

**Comment by ezhulenev** [comment]
You can use XLA:FFI, see example in jax: https://github.com/google/jax/blob/main/docs/cuda_custom_call/foo.cu.cc#L71. You can get access to buffer dimensions via `xla::ffi::Buffer` argument, however it is not strided in a sense that it is always a dense buffer (XLA never supported truly strided buffers).

**Comment by cheshire** [comment]
> XLA never supported truly strided buffers

Should we support them? In a sense that custom call gets a "stride" argument? And XLA generates such "strided" arguments after e.g. slice?

**Comment by ezhulenev** [comment]
Yes, that is something not too hard to add on top of dynamic slice fusion, however I guess most of the custom calls are not prepared to handle strides, but this can be done with explicit opt-in.

**Comment by cheshire** [comment]
Yeah exactly, there should be a way to register such custom calls.

**Comment by penumallas1261** [comment]
Hi @mars1248 ,

Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ?

**Comment by penumallas1261** [comment]
Hi @mars1248 ,

Closing the issue due to inactivity for over 3 months, Please feel free to reopen the issue if you still have a concern. Thanks!
